### DISCUSSION TITLE
TicTacToe machine example with actors and TS

### DISCUSSION THREAD

**Discussion by MrNovado** [statelyai/xstate#2874] [RFC]
Hello everyone,

I've made a `TicTacToe` machine with actors implementing Newell and Simon's expert model.
https://github.com/MrNovado/xstate-tic-tac-toe
https://mrnovado.github.io/xstate-tic-tac-toe/

I've always wanted to contribute to `xState` in some way, and there are still not enough examples (especially when it comes to actors) on how to use `xState` to create ~complex machines with `TypeScript`.

So I wanted to ask if this machine is example-worthy and what could be improved to make it so.

Thanks! 😃 

### UPD: 
- Added a configurable state-transition delay for actors, so you could actually see what they are doing in the inspector frame on the right
- Made a little code-style refactoring so everything would look a little more consistent
- Added emotion to make presentation easier to follow
- Added images and a gif to readme to show machines and how-to switch b/w them

**Comment by ChrisShank** [comment]
Very cool!

**Reply by MrNovado** [reply]
Thanks! 😄 
I'll add tests as well; always wanted to try `@xstate/test` and see what it could offer.
Also looking at how I could delay actor' state transitions, so that it'd be easier to see what's they are doing in inspector frame.

**Reply by davidkpiano** [reply]
> Also looking at how I could delay actor' state transitions, so that it'd be easier to see what's they are doing in inspector frame.

Can't do that yet, but advanced simulation is planned!

**Reply by MrNovado** [reply]
> Can't do that yet, but advanced simulation is planned!

Ha, that's nice to hear 😄 
Though, I've found switching from `always` to `after` on state-transition is the next best thing when it comes to adding a configurable delay.

Related commit:
https://github.com/MrNovado/xstate-tic-tac-toe/commit/c580d5bf3d6c7a0b26f90f3d9b4374f4346179a8

Updated demo:
https://mrnovado.github.io/xstate-tic-tac-toe/

Works well enough for now I suppose 😃

**Comment by huan** [comment]
Love this example so much!

I also vote for the _advanced simulation_ for better visualization of the state chart transitions for a better insight!
